clients/cli/src/unwrap.rs

```diff
@@ -0,0 +1,231 @@
+    /// queried if not provided.
+    #[clap(long, value_parser = parse_token_program)]
+    pub unwrapped_token_program: Option<Pubkey>,
+
+    #[clap(long, value_parser = value_parser!(Hash))]
+    pub blockhash: Option<Hash>,
+}
+
+#[serde_as]
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct UnwrapOutput {
+    #[serde_as(as = "DisplayFromStr")]
+    pub unwrapped_mint_address: Pubkey,
+
+    #[serde_as(as = "DisplayFromStr")]
+    pub unwrapped_token_program: Pubkey,
+
+    #[serde_as(as = "DisplayFromStr")]
+    pub recipient_token_account: Pubkey,
+
+    pub amount: u64,
+
+    pub signatures: Vec<Signature>,
+}
+
+impl Display for UnwrapOutput {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        writeln_name_value(
+            f,
+            "Unwrapped token program:",
+            &self.unwrapped_token_program.to_string(),
+        )?;
+        writeln_name_value(
+            f,
+            "Unwrapped mint address:",
+            &self.unwrapped_mint_address.to_string(),
+        )?;
+        writeln_name_value(
+            f,
+            "Recipient unwrapped token account:",
+            &self.recipient_token_account.to_string(),
+        )?;
+        writeln_name_value(f, "Amount unwrapped:", &self.amount.to_string())?;
+
+        writeln!(f, "Signers:")?;
+        for signature in &self.signatures {
+            writeln!(f, "  {signature}")?;
+        }
+
+        Ok(())
+    }
+}
+
+impl QuietDisplay for UnwrapOutput {
+    fn write_str(&self, _: &mut dyn std::fmt::Write) -> std::fmt::Result {
+        Ok(())
+    }
+}
+impl VerboseDisplay for UnwrapOutput {}
+
+pub async fn command_unwrap(
+    config: &Config,
+    args: UnwrapArgs,
+    matches: &clap::ArgMatches,
+    wallet_manager: &mut Option<Rc<RemoteWalletManager>>,
+) -> CommandResult {
+    let payer = config.fee_payer()?;
+
+    // Get wrapped token program ID from the owner of the wrapped token account
+    // if not provided
+    let wrapped_token_program_id = if let Some(program_id) = args.wrapped_token_program {
+        program_id
+    } else {
+        get_account_owner(&config.rpc_client, &args.wrapped_token_account).await?
+    };
+
+    // Get unwrapped mint from the recipient account if not provided
+    let unwrapped_mint_address = if let Some(mint) = args.unwrapped_mint {
+        mint
+    } else {
+        get_mint(&config.rpc_client, &args.unwrapped_token_recipient).await?
+    };
+
+    // Get unwrapped token program ID from the owner of the escrow account
+    // if not provided
+    let unwrapped_token_program = if let Some(program_id) = args.unwrapped_token_program {
+        program_id
+    } else {
+        get_account_owner(&config.rpc_client, &args.escrow_account).await?
+    };
+
+    println_display(
+        config,
+        format!(
+            "Unwrapping {} tokens from mint {}",
+            args.amount, unwrapped_mint_address
+        ),
+    );
```

Comment:
nit: the wording here was a bit strange to me. Does one unwrap the tokens from the unwrapped mint, or from the wrapped mint? It might be clearest to say "Unwrapping {} tokens from mint {wrapped_mint_address} to {unwrapped_mint_address}" -- what do you think?

Reply:
I see what you mean. Yes, think that makes sense.